test-apps/imodel-transformer/src/Main.ts

```diff
@@ -146,8 +147,8 @@ void (async () => {
         assert(!(args.sourceStartChangesetIndex && args.sourceStartChangesetId), "Pick single way to specify starting changeset");
         if (args.sourceStartChangesetIndex) {
           args.sourceStartChangesetId = await IModelHubUtils.queryChangesetId(requestContext, sourceIModelId, args.sourceStartChangesetIndex);
-        } else {
-          args.sourceStartChangesetIndex = await IModelHubUtils.queryChangesetIndex(requestContext, sourceIModelId, args.sourceStartChangesetId!);
+        } else if (args.sourceStartChangesetId) {

```

Comment:
I was not expecting any logic changes in this PR.  Is this a mistake?  If this is an intentional change in behavior is it tested?

Reply:
I will try removing it, maybe it was transient while I was moving stuff around, but I had a typescript error where it couldn't infer that things had been properly set, so I changed the code to be explicit.

Reply:
alright, it looks like my mistake is that my linter removed an "unnecessary assertion" operator (the `!`) while my types were invalid (implicitly `any`), and I didn't catch that while combing the diff, so I added the condition to the "else" since it was logically redundant. I'll just add the `!` back and remove this change